### PR TITLE
DM-39485: Use redis for work dispatch

### PR DESCRIPTION
**Pull request by mfisherlevine**

### CODE REVIEW

File: python/lsst/rubintv/production/redisUtils.py
Comment (TallJimbo) [inline python/lsst/rubintv/production/redisUtils.py:147]: Some documentation of the Redis schema would be really useful: e.g. keys, types, and what they mean.  Absent that, I can't really claim I understood how various states for workers are represented and hence how different methods interact.

**Review by TallJimbo — APPROVED** [review]
Lots of comments, some minor, some about the future, but really not _that_ many given the PR size.

I did not look too closely at:
- logic where I didn't have enough context/background to understand it (e.g. decoding redis objects, non-redis channels)
- `scripts` (that's all machine-generated, right, for gitops, except `scripts/meta`?)
- places where you already had "TODO" comments that suggested that you wanted to rewrite that code substantially already.

File: config/config_slac_testing.yaml
Comment (mfisherlevine) [reply config/config_slac_testing.yaml]: Thanks, this is all getting refactored, and I've not fixed a lot of this. It's also covered by the test suite (or will be).

File: config/config_usdf_comcamsim.yaml
Comment (mfisherlevine) [reply config/config_usdf_comcamsim.yaml:8]: Also gone.

File: python/lsst/rubintv/production/payloads.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/payloads.py:44]: OK, let's come back to this post OR _4_ 🙃

File: python/lsst/rubintv/production/payloads.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/payloads.py:66]: Yeah, that might be a pair-coding project for us another day, perhaps, I might need to take that plunge eventually.

File: python/lsst/rubintv/production/payloads.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/payloads.py]: Yup, thanks, removed.

File: python/lsst/rubintv/production/pipelineRunning.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/pipelineRunning.py:42]: I'm actually forking off of the regular pipeline in some places, so I think just doing it like this should be OK for now. Let's see how it evolves though.

File: python/lsst/rubintv/production/pipelineRunning.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/pipelineRunning.py:221]: Yup, very much understood, and all the more so as of my problems last night. Adding a note elsewhere to come back to this with you.

File: python/lsst/rubintv/production/pipelineRunning.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/pipelineRunning.py:235]: Thanks, making a note of this for when I add timing stuff 👍

File: python/lsst/rubintv/production/pipelineRunning.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/pipelineRunning.py:256]: I think, on balance, the right thing to do here is just to `raise e` after logging that the task failed, which is, as you say, the same as a `break` (I think).

File: python/lsst/rubintv/production/pipelineRunning.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/pipelineRunning.py:280]: I think that's only the case for now, and even then, I'm not sure. Aggregating data, and sending things to consDB, as well as writing scratch files seem at odds with the task structure, so I'm not sure if I want to go adding these things to tasks. Let's see how it goes over the next little while though.

File: python/lsst/rubintv/production/processingControl.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/processingControl.py:62]: OK, will consider that. Also, I'm not even sure we'll end up using these tbh.

File: python/lsst/rubintv/production/processingControl.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/processingControl.py:200]: Yeah, I actually implemented almost exactly that already, by chance, and have a note to clear up all this cruft soon!

File: python/lsst/rubintv/production/processingControl.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/processingControl.py:398]: Sure, thanks.

File: python/lsst/rubintv/production/processingControl.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/processingControl.py:561]: Yeah, I need to think about this properly. There's a TODO there at least.

File: python/lsst/rubintv/production/processingControl.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/processingControl.py:649]: I totally agree - this is basically abhorrent. I'm putting a big `TODO - consider removing this completely`, and will get to that before ever using it, but will merge it for the time being, as nobody is realistically going to use this any time soon.

File: python/lsst/rubintv/production/processingControl.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/processingControl.py:794]: OK, I think that's fair. I'm filing a new ticket for that though, to unblock the merge, but I think it might just trivially work for all cameras actually, now that I look at it, I just don't have time before OR4 to test that properly.

File: python/lsst/rubintv/production/processingControl.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/processingControl.py:923]: Thanks, that is the official project way 👍

File: python/lsst/rubintv/production/redisUtils.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/redisUtils.py:147]: Yeah, I think that's a good suggestion - putting that on the list for when we look at pydantic stuff 👍

File: python/lsst/rubintv/production/redisUtils.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/redisUtils.py]: Thankfully I saw sense and made everyone just use env vars 🙂

File: python/lsst/rubintv/production/redisUtils.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/redisUtils.py:328]: This is certainly very interesting, and _all_ strikes _me_ as quite ambitious! I'm sticking this on the list as a pair-coding project for us, as I can't address something of this size now, but this is very useful, and won't be lost 🙏

File: python/lsst/rubintv/production/redisUtils.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/redisUtils.py:437]: Yes, that's right, and that's all setup by redis and devOps people - it all Just Works 🙂

File: python/lsst/rubintv/production/watchers.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/watchers.py:41]: Wow, I had never realised that! I need to change a lot of files in that case, but I've done this one for now 😄

File: python/lsst/rubintv/production/redisUtils.py
Comment (mfisherlevine) [reply python/lsst/rubintv/production/redisUtils.py:145]: The short answer is "it's dynamic" 😬 But yes, I take your point - everything needs more documenting I think. Adding it to the list...

**Pull request merged**
